src/rateMailing.go

```diff
@@ -0,0 +1,65 @@
+package src
+
+import (
+	"go_service/src/common"
+	"time"
+)
+
+type SchedulerTimeGateway interface {
+	common.SchedulerReader
+	common.SchedulerWriter
+}
+
+type RateMailer struct {
+	Es common.EmailSender
+	Sr common.SubscriberListReader
+	Sg SchedulerTimeGateway
+	Cr common.CurrencyReader
+}
+
+func (ms RateMailer) Run() {
+	lastTime := ms.Sg.GetLastTime()
+	now := time.Now()
+
+	if (lastTime != nil && lastTime.Day() < now.Day() && lastTime.Hour() >= now.Hour()) || lastTime == nil {
```

Comment:
I don't get this moment "lastTime.Hour() >= now.Hour()", could you explain please? Maybe you need "lastTime.Hour() <= now.Hour()"?

Reply:
Yes, right. Thanks